File: android/sample-ui/src/main/kotlin/com/usesmileid/sampleapps/ui/components/UseSmileIDSampleSectionLabel.kt
package com.usesmileid.sampleapps.ui.components

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.smileid.designsystem.smileLabelSize
import com.smileid.designsystem.smileLabelTracking
import com.usesmileid.sampleapps.ui.theme.UseSmileIDSampleTheme

/** The all-caps group heading above a section. Callers pass the text already cased, so no locale upper-casing. */
@Composable
fun UseSmileIDSampleSectionLabel(
    text: String,
    modifier: Modifier = Modifier,
    testId: String? = null,
) {
    Text(
        text = text,
        // The design's Type/Label, which text-style.overline sets a point small and solid.
        style = UseSmileIDSampleTheme.type.textStyleOverline.copy(
            fontSize = smileLabelSize,
            letterSpacing = smileLabelTracking,
        ),
        color = UseSmileIDSampleTheme.colors.textMuted,
        modifier = modifier.tagged(testId),
    )
}

File: ios/SampleUI/Sources/SampleUI/Components/UseSmileIDSampleSectionLabel.swift
import SwiftUI

/// The all-caps group heading above a section; callers pass the text already cased, so nothing upper-cases per locale.
public struct UseSmileIDSampleSectionLabel: View {
  private let text: String
  private let testId: String?

  @Environment(\.useSmileIDSampleColors) private var colors

  public init(_ text: String, testId: String? = nil) {
    self.text = text
    self.testId = testId
  }

  public var body: some View {
    // The design's Type/Label, which text-style.overline sets a point small and solid.
    UseSmileIDSampleText(
      text,
      style: UseSmileIDSampleTheme.type.textStyleOverline
        .with(size: smileLabelSize, tracking: smileLabelTracking)
    )
    .foregroundColor(colors.textMuted)
    .useSmileIDSampleTestId(testId)
  }
}
